ironfish/src/assert.ts

```diff
@@ -100,4 +100,16 @@ export class Assert {
       throw new Error(message || 'Expected buffers to have the same contents')
     }
   }
+
+  static hasKeys<TObj extends Record<string, unknown>, TKeys extends keyof TObj>(
+    obj: TObj,
+    keys: TKeys[],
+    message?: string,
+  ): asserts obj is TKeys extends keyof TObj ? Required<Pick<TObj, TKeys>> : never {
```

Comment:
Is it possible that `TKeys extends keyof TObj` is `false`? I'm wondering if this ternary is needed at all

Reply:
It's more to convince the type system. Here is the error without the ternary.
```
A type predicate's type must be assignable to its parameter's type.
  Type 'Required<Pick<TObj, TKeys>>' is not assignable to type 'TObj'.
    'Required<Pick<TObj, TKeys>>' is assignable to the constraint of type 'TObj', but 'TObj' could be instantiated with a different subtype of constraint 'Record<string, unknown>'
``` 
I did a lot of debugging and the closest I can get is that TObj cannot be asserted to Required<Pick<TObj, TKeys>> because we need to convince TS that TKeys truly is Keyof TObj. I'm not really sure why.